clerk/middleware.go

```diff
@@ -3,30 +3,63 @@ package clerk
 import (
 	"context"
 	"net/http"
+	"strings"
 )
 
 const (
 	ActiveSession = iota
+	ActiveSessionClaims
 )
 
 func WithSession(client Client) func(handler http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
-		return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
-			session, err := client.Verification().Verify(request)
-			if err != nil {
-				writer.WriteHeader(400)
-				writer.Write([]byte(err.Error()))
-				return
-			}
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if token, isAuthV2 := isAuthV2Request(r, client); isAuthV2 {
+				// Validate using session token
+				claims, err := client.VerifyToken(token)
+				if err != nil {
+					w.WriteHeader(http.StatusUnauthorized)
+					_, _ = w.Write([]byte(err.Error()))
+					return
+				}
+
+				ctx := context.WithValue(r.Context(), ActiveSessionClaims, claims)
+				next.ServeHTTP(w, r.WithContext(ctx))
+			} else {
+				// Validate using session verify request
+				session, err := client.Verification().Verify(r)
+				if err != nil {
+					w.WriteHeader(http.StatusUnauthorized)
```

Comment:
Before we returned a 400 Bad Request if the verification failed. Do you think we should change it to a 401 Unauthorized as more appropriate status code?